Rely on Go Modules to ensure package dependencies and replicable builds

# why we need it?

Go Modules is the official Go dependency management tool, and it’s directly implemented into the go toolchain.
We need this in order to:
- Automatically detect import statements and then update dependencies graph, lock, download, and install appropriate versions of packages.
[
Quoting golang official docs](https://github.com/golang/go/wiki/Modules#go-modules):

> In Go 1.14, module support is considered ready for production use, and all users are encouraged to migrate to modules from other dependency management systems.

Also[ quoting the golang official blog](https://blog.golang.org/using-go-modules), the reason why this does not break compatibility with previous go versions:

> (Inside $GOPATH/src, for compatibility, **the go command still runs in the old GOPATH mode, even if a go.mod is found**. See the go command documentation for details.) Starting in Go 1.13, module mode will be the default for all development.

# minimum specified version

we will need to ensure we have a go.mod on all go clients ( and test at least with the minimum 1.12 go version to say we support >=1.12 ). Ideally it should be 1.11 but redigo outputs an error:
```
$ gvm use go1.11
Now using version go1.11
filipe@filipe-ThinkPad-T490:~/go/src/github.com/RedisBloom/redisbloom-go$ GOCMD="GO111MODULE=on go" make
GO111MODULE=on go get -t -v ./...
go get: -t flag is a no-op when using modules
github.com/gomodule/redigo/redis
go build github.com/gomodule/redigo/redis: module requires Go 1.14
make: *** [Makefile:24: get] Error 1
```

--- a/.circleci/config.yml
+++ b/.circleci/config.yml
@@ -18,7 +18,7 @@ commands:
 jobs:
   build:
     docker:
-      - image: circleci/golang:1.9
+      - image: circleci/golang:1.12
       
       - image: redislabs/redisgraph:edge
 

--- a/.gitignore
+++ b/.gitignore
@@ -1,6 +1,8 @@
 *
 !/**/
 !**/*.go
+!**/go.mod
+!**/go.sum
 !.gitignore
 !.circleci/config.yml
 !/tests/*.bz2

--- a/Makefile
+++ b/Makefile
@@ -1,5 +1,5 @@
 # Go parameters
-GOCMD=go
+GOCMD=GO111MODULE=on go
 GOBUILD=$(GOCMD) build
 GOINSTALL=$(GOCMD) install
 GOCLEAN=$(GOCMD) clean

--- a/go.mod
+++ b/go.mod
@@ -1,0 +1,9 @@
+module github.com/RedisGraph/redisgraph-go
+
+go 1.12
+
+require (
+	github.com/gomodule/redigo v2.0.0+incompatible
+	github.com/olekukonko/tablewriter v0.0.4
+	github.com/stretchr/testify v1.6.1
+)

--- a/go.mod
+++ b/go.mod
@@ -3,7 +3,7 @@ module github.com/RedisGraph/redisgraph-go
 go 1.12
 
 require (
-	github.com/gomodule/redigo v2.0.0+incompatible
+	github.com/gomodule/redigo v1.8.2
 	github.com/olekukonko/tablewriter v0.0.4
 	github.com/stretchr/testify v1.6.1
 )

